use  qute quarkus 2.1.0 dev run error

### Describe the bug

SLF4J: Class path contains multiple SLF4J bindings.
SLF4J: Found binding in [jar:file:/Applications/SpringToolSuite4.app/Contents/Eclipse/plugins/org.eclipse.m2e.maven.runtime.slf4j.simple_1.18.0.20210402-1458/jars/slf4j-simple-1.7.5.jar!/org/slf4j/impl/StaticLoggerBinder.class]
SLF4J: Found binding in [file:/Applications/SpringToolSuite4.app/Contents/Eclipse/configuration/org.eclipse.osgi/5/0/.cp/org/slf4j/impl/StaticLoggerBinder.class]
SLF4J: See http://www.slf4j.org/codes.html#multiple_bindings for an explanation.
SLF4J: Actual binding is of type [org.slf4j.impl.SimpleLoggerFactory]
SLF4J: Class path contains multiple SLF4J bindings.
SLF4J: Found binding in [jar:file:/Applications/SpringToolSuite4.app/Contents/Eclipse/plugins/org.eclipse.m2e.maven.runtime.slf4j.simple_1.18.0.20210402-1458/jars/slf4j-simple-1.7.5.jar!/org/slf4j/impl/StaticLoggerBinder.class]
SLF4J: Found binding in [file:/Applications/SpringToolSuite4.app/Contents/Eclipse/configuration/org.eclipse.osgi/5/0/.cp/org/slf4j/impl/StaticLoggerBinder.class]
SLF4J: See http://www.slf4j.org/codes.html#multiple_bindings for an explanation.
SLF4J: Actual binding is of type [org.slf4j.impl.SimpleLoggerFactory]
[INFO] Scanning for projects...
[INFO] 
[INFO] [1m-------------------< [0;36mcom.weir.blog:weirblog-quarkus[0;1m >-------------------[m
[INFO] [1mBuilding weirblog-quarkus 2.0[m
[INFO] [1m--------------------------------[ jar ]---------------------------------[m
[INFO] 
[INFO] [1m--- [0;32mquarkus-maven-plugin:2.1.1.Final:generate-code[m [1m(default)[m @ [36mweirblog-quarkus[0;1m ---[m
[INFO] 
[INFO] [1m--- [0;32mmaven-resources-plugin:2.6:resources[m [1m(default-resources)[m @ [36mweirblog-quarkus[0;1m ---[m
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] Copying 2476 resources
[INFO] 
[INFO] [1m--- [0;32mmaven-compiler-plugin:3.8.1:compile[m [1m(default-compile)[m @ [36mweirblog-quarkus[0;1m ---[m
[INFO] Changes detected - recompiling the module!
[INFO] Compiling 41 source files to /Users/weir/git/weirblog-quarkus/weirblog-quarkus/target/classes
[INFO] /Users/weir/git/weirblog-quarkus/weirblog-quarkus/src/main/java/com/weirblog/util/ImageUtils.java: /Users/weir/git/weirblog-quarkus/weirblog-quarkus/src/main/java/com/weirblog/util/ImageUtils.java使用或覆盖了已过时的 API。
[INFO] /Users/weir/git/weirblog-quarkus/weirblog-quarkus/src/main/java/com/weirblog/util/ImageUtils.java: 有关详细信息, 请使用 -Xlint:deprecation 重新编译。
[INFO] 
[INFO] [1m--- [0;32mquarkus-maven-plugin:2.1.1.Final:dev[m [1m(default-cli)[m @ [36mweirblog-quarkus[0;1m ---[m
[INFO] Invoking org.apache.maven.plugins:maven-resources-plugin:2.6:testResources @ weirblog-quarkus
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] skip non existing resourceDirectory /Users/weir/git/weirblog-quarkus/weirblog-quarkus/src/test/resources
[INFO] Invoking org.apache.maven.plugins:maven-compiler-plugin:3.8.1:testCompile @ weirblog-quarkus
[INFO] Changes detected - recompiling the module!
[INFO] Compiling 1 source file to /Users/weir/git/weirblog-quarkus/weirblog-quarkus/target/test-classes
Listening for transport dt_socket at address: 5005
Press [h] for more options>
Tests paused
Press [r] to resume testing, [h] for more options>
Press [r] to resume testing, [o] Toggle test output, [h] for more options>
[[34;1mINFO[m] Checking for existing resources in: /Users/weir/git/weirblog-quarkus/weirblog-quarkus/src/main/kubernetes.
2021-08-05 09:08:51,675 ERROR [io.qua.dep.dev.IsolatedDevModeMain] (main) Failed to start quarkus: java.lang.RuntimeException: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
	[error]: Build step io.quarkus.qute.deployment.QuteProcessor#analyzeTemplates threw an exception: java.lang.IndexOutOfBoundsException: Index: -1
	at java.base/java.util.Collections$EmptyList.get(Collections.java:4483)
	at io.quarkus.qute.Parser.parseExpression(Parser.java:724)
	at io.quarkus.qute.Parser.createPart(Parser.java:762)
	at io.quarkus.qute.Parser.parseExpression(Parser.java:735)
	at io.quarkus.qute.Parser.apply(Parser.java:815)
	at io.quarkus.qute.Parser.flushTag(Parser.java:486)
	at io.quarkus.qute.Parser.tag(Parser.java:293)
	at io.quarkus.qute.Parser.processCharacter(Parser.java:200)
	at io.quarkus.qute.Parser.parse(Parser.java:132)
	at io.quarkus.qute.EngineImpl.load(EngineImpl.java:138)
	at java.base/java.util.concurrent.ConcurrentHashMap.computeIfAbsent(ConcurrentHashMap.java:1705)
	at io.quarkus.qute.EngineImpl.getTemplate(EngineImpl.java:116)
	at io.quarkus.qute.deployment.QuteProcessor.analyzeTemplates(QuteProcessor.java:459)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at io.quarkus.deployment.ExtensionLoader$2.execute(ExtensionLoader.java:820)
	at io.quarkus.builder.BuildContext.run(BuildContext.java:277)
	at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2442)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1476)
	at java.base/java.lang.Thread.run(Thread.java:834)
	at org.jboss.threads.JBossThread.run(JBossThread.java:501)

	at io.quarkus.runner.bootstrap.AugmentActionImpl.runAugment(AugmentActionImpl.java:415)
	at io.quarkus.runner.bootstrap.AugmentActionImpl.createInitialRuntimeApplication(AugmentActionImpl.java:275)
	at io.quarkus.runner.bootstrap.AugmentActionImpl.createInitialRuntimeApplication(AugmentActionImpl.java:66)
	at io.quarkus.deployment.dev.IsolatedDevModeMain.firstStart(IsolatedDevModeMain.java:91)
	at io.quarkus.deployment.dev.IsolatedDevModeMain.accept(IsolatedDevModeMain.java:424)
	at io.quarkus.deployment.dev.IsolatedDevModeMain.accept(IsolatedDevModeMain.java:64)
	at io.quarkus.bootstrap.app.CuratedApplication.runInCl(CuratedApplication.java:136)
	at io.quarkus.bootstrap.app.CuratedApplication.runInAugmentClassLoader(CuratedApplication.java:93)
	at io.quarkus.deployment.dev.DevModeMain.start(DevModeMain.java:145)
	at io.quarkus.deployment.dev.DevModeMain.main(DevModeMain.java:63)
Caused by: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
	[error]: Build step io.quarkus.qute.deployment.QuteProcessor#analyzeTemplates threw an exception: java.lang.IndexOutOfBoundsException: Index: -1
	at java.base/java.util.Collections$EmptyList.get(Collections.java:4483)
	at io.quarkus.qute.Parser.parseExpression(Parser.java:724)
	at io.quarkus.qute.Parser.createPart(Parser.java:762)
	at io.quarkus.qute.Parser.parseExpression(Parser.java:735)
	at io.quarkus.qute.Parser.apply(Parser.java:815)
	at io.quarkus.qute.Parser.flushTag(Parser.java:486)
	at io.quarkus.qute.Parser.tag(Parser.java:293)
	at io.quarkus.qute.Parser.processCharacter(Parser.java:200)
	at io.quarkus.qute.Parser.parse(Parser.java:132)
	at io.quarkus.qute.EngineImpl.load(EngineImpl.java:138)
	at java.base/java.util.concurrent.ConcurrentHashMap.computeIfAbsent(ConcurrentHashMap.java:1705)
	at io.quarkus.qute.EngineImpl.getTemplate(EngineImpl.java:116)
	at io.quarkus.qute.deployment.QuteProcessor.analyzeTemplates(QuteProcessor.java:459)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at io.quarkus.deployment.ExtensionLoader$2.execute(ExtensionLoader.java:820)
	at io.quarkus.builder.BuildContext.run(BuildContext.java:277)
	at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2442)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1476)
	at java.base/java.lang.Thread.run(Thread.java:834)
	at org.jboss.threads.JBossThread.run(JBossThread.java:501)

	at io.quarkus.builder.Execution.run(Execution.java:116)
	at io.quarkus.builder.BuildExecutionBuilder.execute(BuildExecutionBuilder.java:79)
	at io.quarkus.deployment.QuarkusAugmentor.run(QuarkusAugmentor.java:151)
	at io.quarkus.runner.bootstrap.AugmentActionImpl.runAugment(AugmentActionImpl.java:413)
	... 9 more
Caused by: java.lang.IndexOutOfBoundsException: Index: -1
	at java.base/java.util.Collections$EmptyList.get(Collections.java:4483)
	at io.quarkus.qute.Parser.parseExpression(Parser.java:724)
	at io.quarkus.qute.Parser.createPart(Parser.java:762)
	at io.quarkus.qute.Parser.parseExpression(Parser.java:735)
	at io.quarkus.qute.Parser.apply(Parser.java:815)
	at io.quarkus.qute.Parser.flushTag(Parser.java:486)
	at io.quarkus.qute.Parser.tag(Parser.java:293)
	at io.quarkus.qute.Parser.processCharacter(Parser.java:200)
	at io.quarkus.qute.Parser.parse(Parser.java:132)
	at io.quarkus.qute.EngineImpl.load(EngineImpl.java:138)
	at java.base/java.util.concurrent.ConcurrentHashMap.computeIfAbsent(ConcurrentHashMap.java:1705)
	at io.quarkus.qute.EngineImpl.getTemplate(EngineImpl.java:116)
	at io.quarkus.qute.deployment.QuteProcessor.analyzeTemplates(QuteProcessor.java:459)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at io.quarkus.deployment.ExtensionLoader$2.execute(ExtensionLoader.java:820)
	at io.quarkus.builder.BuildContext.run(BuildContext.java:277)
	at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2442)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1476)
	at java.base/java.lang.Thread.run(Thread.java:834)
	at org.jboss.threads.JBossThread.run(JBossThread.java:501)

2021-08-05 09:08:51,676 INFO  [io.qua.dep.dev.IsolatedDevModeMain] (main) Attempting to start live reload endpoint to recover from previous Quarkus startup failure
2021-08-05 09:08:51,898 INFO  [org.jbo.threads] (main) JBoss Threads version 3.4.0.Final
2021-08-05 09:08:55,838 WARN  [io.ver.cor.imp.BlockedThreadChecker] (vertx-blocked-thread-checker) Thread Thread[vert.x-eventloop-thread-22,5,main] has been blocked for 3864 ms, time limit is 2000 ms: io.vertx.core.VertxException: Thread blocked
	at io.vertx.core.net.impl.TCPServerBase.listen(TCPServerBase.java:97)
	at io.vertx.core.http.impl.HttpServerImpl.listen(HttpServerImpl.java:220)
	at io.vertx.core.http.impl.HttpServerImpl.listen(HttpServerImpl.java:150)
	at io.vertx.core.http.impl.HttpServerImpl.listen(HttpServerImpl.java:155)
	at io.quarkus.vertx.http.runtime.VertxHttpRecorder$WebDeploymentVerticle.setupTcpHttpServer(VertxHttpRecorder.java:991)
	at io.quarkus.vertx.http.runtime.VertxHttpRecorder$WebDeploymentVerticle.start(VertxHttpRecorder.java:933)
	at io.vertx.core.impl.DeploymentManager.lambda$doDeploy$5(DeploymentManager.java:196)
	at io.vertx.core.impl.DeploymentManager$$Lambda$1229/0x0000000840698440.handle(Unknown Source)
	at io.vertx.core.impl.AbstractContext.dispatch(AbstractContext.java:96)
	at io.vertx.core.impl.AbstractContext.dispatch(AbstractContext.java:59)
	at io.vertx.core.impl.EventLoopContext.lambda$runOnContext$0(EventLoopContext.java:37)
	at io.vertx.core.impl.EventLoopContext$$Lambda$1230/0x0000000840699040.run(Unknown Source)
	at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:164)
	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:472)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:500)
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:989)
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.base@11.0.10/java.lang.Thread.run(Thread.java:834)

2021-08-05 09:08:55,853 WARN  [io.ver.cor.imp.BlockedThreadChecker] (vertx-blocked-thread-checker) Thread Thread[vert.x-eventloop-thread-13,5,main] has been blocked for 3870 ms, time limit is 2000 ms: io.vertx.core.VertxException: Thread blocked
	at java.base@11.0.10/java.net.Inet6AddressImpl.lookupAllHostAddr(Native Method)
	at java.base@11.0.10/java.net.InetAddress$PlatformNameService.lookupAllHostAddr(InetAddress.java:929)
	at java.base@11.0.10/java.net.InetAddress.getAddressesFromNameService(InetAddress.java:1519)
	at java.base@11.0.10/java.net.InetAddress$NameServiceAddresses.get(InetAddress.java:848)
	at java.base@11.0.10/java.net.InetAddress.getAllByName0(InetAddress.java:1509)
	at java.base@11.0.10/java.net.InetAddress.getLocalHost(InetAddress.java:1641)
	at java.management@11.0.10/sun.management.VMManagementImpl.getVmId(VMManagementImpl.java:144)
	at java.management@11.0.10/sun.management.RuntimeImpl.getName(RuntimeImpl.java:59)
	at java.base@11.0.10/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base@11.0.10/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base@11.0.10/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base@11.0.10/java.lang.reflect.Method.invoke(Method.java:566)
	at io.netty.channel.DefaultChannelId.defaultProcessId(DefaultChannelId.java:120)
	at io.netty.channel.DefaultChannelId.<clinit>(DefaultChannelId.java:77)
	at io.netty.channel.AbstractChannel.newId(AbstractChannel.java:113)
	at io.netty.channel.AbstractChannel.<init>(AbstractChannel.java:73)
	at io.netty.channel.nio.AbstractNioChannel.<init>(AbstractNioChannel.java:80)
	at io.netty.channel.nio.AbstractNioMessageChannel.<init>(AbstractNioMessageChannel.java:42)
	at io.netty.channel.socket.nio.NioServerSocketChannel.<init>(NioServerSocketChannel.java:89)
	at io.netty.channel.socket.nio.NioServerSocketChannel.<init>(NioServerSocketChannel.java:75)
	at io.vertx.core.net.impl.transport.Transport$$Lambda$1283/0x00000008406b7840.newChannel(Unknown Source)
	at io.netty.bootstrap.AbstractBootstrap.initAndRegister(AbstractBootstrap.java:310)
	at io.netty.bootstrap.AbstractBootstrap.doBind(AbstractBootstrap.java:272)
	at io.netty.bootstrap.AbstractBootstrap.bind(AbstractBootstrap.java:268)
	at io.vertx.core.net.impl.AsyncResolveConnectHelper.lambda$doBind$2(AsyncResolveConnectHelper.java:58)
	at io.vertx.core.net.impl.AsyncResolveConnectHelper$$Lambda$1284/0x00000008406b7c40.handle(Unknown Source)
	at io.vertx.core.impl.future.FutureImpl$3.onSuccess(FutureImpl.java:124)
	at io.vertx.core.impl.future.FutureBase.emitSuccess(FutureBase.java:62)
	at io.vertx.core.impl.future.FutureImpl.addListener(FutureImpl.java:164)
	at io.vertx.core.impl.future.FutureImpl.onComplete(FutureImpl.java:132)
	at io.vertx.core.impl.AddressResolver.resolveHostname(AddressResolver.java:86)
	at io.vertx.core.impl.VertxImpl.resolveAddress(VertxImpl.java:837)
	at io.vertx.core.net.impl.AsyncResolveConnectHelper.doBind(AsyncResolveConnectHelper.java:73)
	at io.vertx.core.net.impl.TCPServerBase.listen(TCPServerBase.java:130)
	at io.vertx.core.http.impl.HttpServerImpl.listen(HttpServerImpl.java:220)
	at io.vertx.core.http.impl.HttpServerImpl.listen(HttpServerImpl.java:150)
	at io.vertx.core.http.impl.HttpServerImpl.listen(HttpServerImpl.java:155)
	at io.quarkus.vertx.http.runtime.VertxHttpRecorder$WebDeploymentVerticle.setupTcpHttpServer(VertxHttpRecorder.java:991)
	at io.quarkus.vertx.http.runtime.VertxHttpRecorder$WebDeploymentVerticle.start(VertxHttpRecorder.java:933)
	at io.vertx.core.impl.DeploymentManager.lambda$doDeploy$5(DeploymentManager.java:196)
	at io.vertx.core.impl.DeploymentManager$$Lambda$1229/0x0000000840698440.handle(Unknown Source)
	at io.vertx.core.impl.AbstractContext.dispatch(AbstractContext.java:96)
	at io.vertx.core.impl.AbstractContext.dispatch(AbstractContext.java:59)
	at io.vertx.core.impl.EventLoopContext.lambda$runOnContext$0(EventLoopContext.java:37)
	at io.vertx.core.impl.EventLoopContext$$Lambda$1230/0x0000000840699040.run(Unknown Source)
	at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:164)
	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:472)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:500)
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:989)
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.base@11.0.10/java.lang.Thread.run(Thread.java:834)

2021-08-05 09:08:55,856 WARN  [io.ver.cor.imp.BlockedThreadChecker] (vertx-blocked-thread-checker) Thread Thread[vert.x-eventloop-thread-8,5,main] has been blocked for 3853 ms, time limit is 2000 ms: io.vertx.core.VertxException: Thread blocked
	at io.vertx.core.net.impl.TCPServerBase.listen(TCPServerBase.java:97)
	at io.vertx.core.http.impl.HttpServerImpl.listen(HttpServerImpl.java:220)
	at io.vertx.core.http.impl.HttpServerImpl.listen(HttpServerImpl.java:150)
	at io.vertx.core.http.impl.HttpServerImpl.listen(HttpServerImpl.java:155)
	at io.quarkus.vertx.http.runtime.VertxHttpRecorder$WebDeploymentVerticle.setupTcpHttpServer(VertxHttpRecorder.java:991)
	at io.quarkus.vertx.http.runtime.VertxHttpRecorder$WebDeploymentVerticle.start(VertxHttpRecorder.java:933)
	at io.vertx.core.impl.DeploymentManager.lambda$doDeploy$5(DeploymentManager.java:196)
	at io.vertx.core.impl.DeploymentManager$$Lambda$1229/0x0000000840698440.handle(Unknown Source)
	at io.vertx.core.impl.AbstractContext.dispatch(AbstractContext.java:96)
	at io.vertx.core.impl.AbstractContext.dispatch(AbstractContext.java:59)
	at io.vertx.core.impl.EventLoopContext.lambda$runOnContext$0(EventLoopContext.java:37)
	at io.vertx.core.impl.EventLoopContext$$Lambda$1230/0x0000000840699040.run(Unknown Source)
	at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:164)
	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:472)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:500)
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:989)
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.base@11.0.10/java.lang.Thread.run(Thread.java:834)



### Expected behavior

_No response_

### Actual behavior

_No response_

### How to Reproduce?

_No response_

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

11

### GraalVM version (if different from Java)

11

### Quarkus version or git rev

2.1.x

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

--- a/independent-projects/qute/core/src/main/java/io/quarkus/qute/Parser.java
+++ b/independent-projects/qute/core/src/main/java/io/quarkus/qute/Parser.java
@@ -489,13 +489,17 @@ class Parser implements Function<String, Expression>, ParserHelper {
     }
 
     private TemplateException parserError(String message) {
+        return parserError(message, origin(0));
+    }
+
+    private static TemplateException parserError(String message, Origin origin) {
         StringBuilder builder = new StringBuilder("Parser error");
-        if (!templateId.equals(generatedId)) {
-            builder.append(" in template [").append(templateId).append("]");
+        if (!origin.getTemplateId().equals(origin.getTemplateGeneratedId())) {
+            builder.append(" in template [").append(origin.getTemplateId()).append("]");
         }
-        builder.append(" on line ").append(line).append(": ")
+        builder.append(" on line ").append(origin.getLine()).append(": ")
                 .append(message);
-        return new TemplateException(origin(0),
+        return new TemplateException(origin,
                 builder.toString());
     }
 
@@ -719,6 +723,10 @@ class Parser implements Function<String, Expression>, ParserHelper {
             }
         }
 
+        if (strParts.isEmpty()) {
+            throw parserError("empty expression found {" + value + "}", origin);
+        }
+
         // Safe expressions
         int lastIdx = strParts.size() - 1;
         String last = strParts.get(lastIdx);
@@ -734,13 +742,7 @@ class Parser implements Function<String, Expression>, ParserHelper {
         while (strPartsIterator.hasNext()) {
             Part part = createPart(idGenerator, namespace, first, strPartsIterator, scope, origin);
             if (!isValidIdentifier(part.getName())) {
-                StringBuilder builder = new StringBuilder("Invalid identifier found [");
-                builder.append(value).append("]");
-                if (!origin.getTemplateId().equals(origin.getTemplateGeneratedId())) {
-                    builder.append(" in template [").append(origin.getTemplateId()).append("]");
-                }
-                builder.append(" on line ").append(origin.getLine());
-                throw new TemplateException(builder.toString());
+                throw parserError("invalid identifier found {" + value + "}", origin);
             }
             if (first == null) {
                 first = part;

--- a/independent-projects/qute/core/src/test/java/io/quarkus/qute/ParserTest.java
+++ b/independent-projects/qute/core/src/test/java/io/quarkus/qute/ParserTest.java
@@ -252,7 +252,7 @@ public class ParserTest {
             engine.parse("{foo\nfoo}");
             fail();
         } catch (Exception expected) {
-            assertTrue(expected.getMessage().contains("Invalid identifier found"), expected.toString());
+            assertEquals("Parser error on line 1: invalid identifier found {foo\nfoo}", expected.getMessage());
         }
     }
 
@@ -347,6 +347,12 @@ public class ParserTest {
         Expression nameToUpperCase = find(expressions, "name.toUpperCase");
         assertExpr(expressions, "upperCase.length", 2, "upperCase<set#" + nameToUpperCase.getGeneratedId() + ">.length");
         assertEquals(":3:5", loopLetLoopLet.data("foo", new Foo()).render());
+    }
+
+    @Test
+    public void testInvalidNamespaceExpression() {
+        assertParserError("{data: }",
+                "Parser error on line 1: empty expression found {data:}", 1);
     }
 
     public static class Foo {

